conjur/host/host_controller.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+
+"""
+HostController module
+
+This module is the controller that facilitates all host actions
+"""
+
+# Builtins
+import sys
+
+class HostController():
+    """
+    HostController
+
+    This class represents the Presentation Layer for the User command.
+    """
+    def __init__(self, client, host_resource_data):
+        self.client=client
+        self.host_resource_data=host_resource_data
+
+    def rotate_api_key(self, resource):
+        """
+        Method that distinguishes between the type of actions
+        and facilitates the calls accordingly
+        """
+        self.prompt_for_host_id()
+        new_api_key = self.client.rotate_anothers_api_key(resource,
+                                                          self.host_resource_data.host_to_update)
+        sys.stdout.write(f"API key for '{self.host_resource_data.host_to_update}' " \
+                        f"was successfully rotated. New API key is: {new_api_key}\n")
+
+    def prompt_for_host_id(self):
```

Comment:
general.. incase the function can not raise a prompt (case host_to_update is None) should add the if_needed suffix prompt_for_host_id-> prompt_for_host_id_if_needed